test/e2e/util/hypershift_framework.go

```diff
@@ -322,3 +324,18 @@ func (h *hypershiftTest) teardownHostedCluster(ctx context.Context, hc *hyperv1.
 		}
 	}
 }
+
+func (h *hypershiftTest) summarizeHCConditions(hostedCluster *hyperv1.HostedCluster) {
```

Comment:
can you provide a before vs after output? for when the image rollout doesn't complete like in the original ticket https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-storage-operator/403/pull-ci-openshift-cluster-storage-operator-master-hypershift-aws-e2e-external/1709056236764270592

Reply:
@enxebre  Ive added an output for before and after that prints out all the conditions in the latest commit. I don't think that is the cleanest way, are we okay to only log the conditions that have changed after the image rollout ?

Reply:
I'm sorry, with before vs after output I meant that today if image rollout fails we get https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-storage-operator/403/pull-ci-openshift-cluster-storage-operator-master-hypershift-aws-e2e-external/1709056236764270592
So basically
```
status.version.history[0].state is "Partial", but we want "Completed"
    util.go:316: Waiting for hostedcluster rollout. Image: registry.build05.ci.openshift.org/ci-op-89mjfjx7/release@sha256:007fb115c67af7570e4f9511f88eddf22fd46059c2315ce1fc193d288095ece2: status.version.history[0].state is "Partial", but we want "Completed"
    util.go:316: Waiting for hostedcluster rollout. Image: registry.build05.ci.openshift.org/ci-op-89mjfjx7/release@sha256:007fb115c67af7570e4f9511f88eddf22fd46059c2315ce1fc193d288095ece2: status.version.history[0].state is "Partial", but we want "Completed"
    util.go:316: Waiting for hostedcluster rollout. Image: registry.build05.ci.openshift.org/ci-op-89mjfjx7/release@sha256:007fb115c67af7570e4f9511f88eddf22fd46059c2315ce1fc193d288095ece2: status.version.history[0].state is "Partial", but we want "Completed"
    util.go:321: 
        failed waiting for hostedcluster image rollout: status.version.history[0].state is "Partial", but we want "Completed"
        Unexpected error:
            <*errors.errorString | 0xc0002f3ef0>: 
            timed out waiting for the condition
            {
                s: "timed out waiting for the condition",
            }
        occurred
    --- FAIL: TestCreateClusterProxy/ValidateHostedCluster (2382.36s)
}
```

Can you illustrate in this PR desc what we would get with your changes so it's easier to understand the value at a glance for any reviewer

Reply:
Ah I get you, let me run it locally to get an example output.

Reply:
Another thing we can take the opportunity to review is the meaningless logs in loop like in the case above ` util.go:316: Waiting for hostedcluster rollout. Image: registry.build05.ci.openshift.org/ci-op-89mjfjx7/release@sha256:007fb115c67af7570e4f9511f88eddf22fd46059c2315ce1fc193d288095ece2: status.version.history[0].state is "Partial", but we want "Completed"` or https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_hypershift/3193/pull-ci-openshift-hypershift-main-e2e-aws/1725350700231168000

Reply:
If we keep track of the last state we saw, we can log only on edge transitions, which will keep the noise down.